tests/integration/module.test.js

```diff
@@ -0,0 +1,214 @@
+
+    const actualFee = fees[bip]
+
+    expect(abs(initialBalance0 - finalBalance0 - TRANSACTION.value - actualFee)).toBeLessThanOrEqual(1n)
+
+    expect(finalBalance1).toBe(initialBalance1 + TRANSACTION.value)
+  })
+
+  test('should derive an account, sign a message and verify its signature', async () => {
+    const account0 = await wallet.getAccount(2)
+
+    const message = 'Hello, world!'
+
+    const signature = await account0.sign(message)
+    const verified = await account0.verify(message, signature)
+    expect(verified).toBe(true)
+  })
+
+  test('should derive an account, send some transactions and retrieve the proper transfer history', async () => {
+    const account0 = await wallet.getAccount(2)
+    const account1 = await wallet.getAccount(3)
+
+    const address0 = await account0.getAddress()
+    const address1 = await account1.getAddress()
+
+    await account0.sendTransaction({ to: address1, value: 1_000n })
+    await waiter.mine()
+
+    await account1.sendTransaction({ to: address0, value: 800n })
+    await waiter.mine() 
+
+    await account0.sendTransaction({ to: address1, value: 2_000n })
+    await waiter.mine()
+
+    const transfers0All = await account0.getTransfers()
+
+    const recent = transfers0All.slice(-3)
+    expect(recent).toHaveLength(3)
+
+    const [tx1, tx2, tx3] = recent
+
+    expect(tx1.direction).toBe('outgoing')
+    expect(tx1.recipient).toBe(address1)
+    expect(tx1.value).toBe(1000n)
+
+    expect(tx2.direction).toBe('incoming')
+    expect(tx2.address).toBe(address0)
+    expect(tx2.value).toBe(800n)
+
+    expect(tx3.direction).toBe('outgoing')
+    expect(tx3.recipient).toBe(address1)
+    expect(tx3.value).toBe(2000n)
+  })
+
+  test('should get a max spendable amount that is actually spendable', async () => {
+    const account0 = await wallet.getAccount(2)
+    const account1 = await wallet.getAccount(3)
+
+    const balance = await account0.getBalance()
+    const { amount } = await account0.getMaxSpendable()
+
+    expect(amount).toBeGreaterThan(0n)
+    expect(amount).toBeLessThan(balance)
+
+    const TRANSACTION = {
+      to: await account1.getAddress(),
+      value: amount
+    }
+
+    const { hash } = await account0.sendTransaction(TRANSACTION)
+
+    await waiter.mine()
+
+    const transaction = parseRawTransaction(bitcoin.getRawTransaction(hash), TRANSACTION.to)
+    expect(transaction.txid).toBe(hash)
+    expect(transaction.details[0].address).toBe(TRANSACTION.to)
+  })
+
+  test('should dispose the wallet and erase the private keys of the accounts', async () => {
+    const account0 = await wallet.getAccount(2)
+    const account1 = await wallet.getAccount(3)
+
+    wallet.dispose()
+
+    const MESSAGE = 'Hello, world!'
+
+    for (const account of [account0, account1]) {
+      expect(() => account.keyPair.privateKey).toThrow()
+      await expect(account.sendTransaction({ to: await account.getAddress(), value: 1_000n })).rejects.toThrow()
```

Comment:
Assert against an error message like you are doing for the 'sign' method. You can also use regex if the message is too long:
```suggestion
      await expect(account.sendTransaction({ to: await account.getAddress(), value: 1_000n })).rejects.toThrow(/Your error message.../)
```

Reply:
It's ok to just assert that it throws without requiring a specific error message.